### ISSUE TITLE
Unable to get this working

### ISSUE THREAD

**Issue by F4u5t** [MicrosoftDocs/vsts-rest-api-specs#140]
can we get a better example?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3820511c-6eab-cad8-7ac3-f8ec8ae5bd25
* Version Independent ID: 2078122b-0830-464b-a958-a49e3eac1e67
* Content: [Attachments (Work Item Tracking)](https://docs.microsoft.com/en-us/rest/api/azure/devops/wit/attachments?view=azure-devops-rest-5.0)
* Content Source: [docs-ref-autogen/5.0/wit/Attachments.yml](https://github.com/MicrosoftDocs/vsts-docs-rest-apis/blob/live/docs-ref-autogen/5.0/wit/Attachments.yml)
* Product: **devops**
* GitHub Login: @wnjenkin
* Microsoft Alias: **whjenkin**

**Comment by mmanela** [comment]
@F4u5t I do see issues in the doc site which we will fix. But can you be specific on what the issue you are hitting to make sure we fix the right things.

Also, please take a look at some of our C# examples
https://github.com/Microsoft/vsts-dotnet-samples/blob/master/ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/WorkItemTracking/AttachmentsSample.cs#L49

**Comment by F4u5t** [comment]
I am trying to use this with DevOps, I think that may be my issue. It seems
the old TFS previously used a Attachment Store, that is not available in
DevOps??
I need to learn how to upload and attachment to a work item on my DevOps
installation.

Thanks for your assistance

**Comment by mmanela** [comment]
I am not sure what attachment store is but the REST API from the docs are what you need to use. Did you take a look at those c# examples?

**Comment by F4u5t** [comment]
I think that is where Im getting lost, the C# exmaples do not seem to use
the REST API, it appears they only use:
using Microsoft.TeamFoundation.WorkItemTracking.WebApi;

using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;

I have researched the REST APi a bit and I think it has not worked for me
yet due to a different Authentication method....I have been using Persona
Access Token, which the REST API does not seem to use???

**Comment by mmanela** [comment]
Microsoft.TeamFoundation.WorkItemTracking.WebApi is our REST api. Those are our generated clients (we generate for C#, Python, Node). They make it convenient to authenticate and call our rest apis

**Comment by phirschorn** [comment]
How do you list the attachments from work items in order to obtain the ID?  The only way I can see how to get the ID is to first upload an attachment which doesn't help if it's already attached.

**Comment by manishmoothedath** [comment]
Correct!!, If I have attachments already in DevOps, how do I list the ID's of my work items ?

**Comment by NavinAT** [comment]
Hai @manishmoothedath @phirschorn

You can get the stored attachments using the work item Id.
Url: http://tfs:host/tfs/collection/_apis/wit/workitems?ids=11111,{specify more ids}&$expand=all&api-version=4.1

In the HttpResponse, under the Relations object you will get Url to download the attachments.
